tests/playgrounds/toolbox_categories.js

```diff
@@ -0,0 +1,37 @@
+var logic_category = {
+  "kind": "CATEGORY",
+  "name": "Logic",
+  "categorystyle":"logic_category",
+  "contents": [
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"controls_if\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_compare\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_operation\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_negate\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_boolean\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_null\" disabled=\"true\"></block>",
+    },
+    {
+      "kind": "BLOCK",
+      "blockxml": "<block type=\"logic_ternary\"></block>",
+    }
+  ],
+};
+
+Blockly.registry.register(Blockly.registry.Type.TOOLBOX_CATEGORY, 'logic_category', logic_category);
```

Comment:
I feel like this might be a little confusing. We're telling developers to register blocks (used in the toolbox xml) with: 
``Blockly.Blocks[TYPE] = ...`` whereas they register categories in the same toolbox xml using: 
``Blockly.registry.register(Blockly.registry.Type.TOOLBOX_CATEGORY, TYPE, ...)``.

I feel like a more appropriate pattern would be to use ``Blockly.Categories[TYPE] = ...``, I recognize that goes against the idea of creating a registry, but I feel in this case it would be more confusing to switch the pattern given the same toolbox XML attribute `type` would end up referring to different patterns depending on the xml tag (block / category).

Reply:
Talked about offline. Removing this for now, and will see if it needs to be added later on.